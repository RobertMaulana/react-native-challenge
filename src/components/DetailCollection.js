import React, {Component} from 'react'
import {
  Image
} from 'react-native'
import moment from 'moment'
import { Container, Content, Card, CardItem, Thumbnail, Text, Button, Icon, Left, Body } from 'native-base';

class DetailCollection extends Component {
  constructor(props){
    super(props)
    this.state = {
    }
  }
  static navigationOptions = {
    title: 'Detail',
  }
  render(){
    const { params } = this.props.navigation.state;
    const restaurant = params.collection;

    return (
      <Container>
        <Content>
            <Card style={{ flex: 0 }}>
              <CardItem>
                <Left>
                  <Thumbnail style={{width: 50, height:50}} source={{uri: 'https://media.glassdoor.com/sql/515676/zomato-squarelogo-1466750863527.png'}}/>
                  <Body>
                      <Text>{restaurant.title}</Text>
                      <Text note>{ moment().format("DD MMM YYYY") }</Text>
                  </Body>
                </Left>
              </CardItem>
              <CardItem>
                <Body>
                  <Image style={{ width: 340, height: 200}} source={{uri: restaurant.image_url}}/>
                  <Text>
                      {restaurant.description}
                  </Text>
                  <Button transparent textStyle={{color: '#87838B'}}>
                      <Icon name="logo-facebook" />
                      <Left><Text style={{ marginLeft: 2 ,fontSize: 12}}>Share it</Text></Left>
                  </Button>
                </Body>
              </CardItem>
           </Card>
        </Content>
      </Container>
    )
  }
}

export default DetailCollection
