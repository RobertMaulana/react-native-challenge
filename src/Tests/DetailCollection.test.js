import React from 'react'
import { shallow } from 'enzyme'

import DetailCollection from '../components/DetailCollection'

describe('<DetailCollection />', () => {
  it('should render correctly', () => {
    const detail = shallow(
      <DetailCollection navigation={{
          state: {
            params: { collection: {} }
          }
      }} />
    )
    expect(detail).toMatchSnapshot()
  })
})
