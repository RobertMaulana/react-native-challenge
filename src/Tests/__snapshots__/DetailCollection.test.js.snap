// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DetailCollection /> should render correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Styled(Container)
    virtual={undefined}
>
    <Styled(Content)
        virtual={undefined}
    >
        <Styled(Card)
            style={
                Object {
                    "flex": 0,
                  }
            }
            virtual={undefined}
        >
            <Styled(CardItem)
                virtual={undefined}
            >
                <Styled(Left)
                    virtual={undefined}
                >
                    <Styled(Thumbnail)
                        source={
                            Object {
                                "uri": "https://media.glassdoor.com/sql/515676/zomato-squarelogo-1466750863527.png",
                              }
                        }
                        style={
                            Object {
                                "height": 50,
                                "width": 50,
                              }
                        }
                        virtual={undefined}
                    />
                    <Styled(Body)
                        virtual={undefined}
                    >
                        <Styled(Text)
                            virtual={undefined}
                        />
                        <Styled(Text)
                            note={true}
                            virtual={undefined}
                        >
                            04 May 2017
                        </Styled(Text)>
                    </Styled(Body)>
                </Styled(Left)>
            </Styled(CardItem)>
            <Styled(CardItem)
                virtual={undefined}
            >
                <Styled(Body)
                    virtual={undefined}
                >
                    <Image
                        source={
                            Object {
                                "uri": undefined,
                              }
                        }
                        style={
                            Object {
                                "height": 200,
                                "width": 340,
                              }
                        }
                    />
                    <Styled(Text)
                        virtual={undefined}
                    />
                    <Styled(Button)
                        textStyle={
                            Object {
                                "color": "#87838B",
                              }
                        }
                        transparent={true}
                        virtual={undefined}
                    >
                        <Styled(Icon)
                            name="logo-facebook"
                            virtual={undefined}
                        />
                        <Styled(Left)
                            virtual={undefined}
                        >
                            <Styled(Text)
                                style={
                                    Object {
                                        "fontSize": 12,
                                        "marginLeft": 2,
                                      }
                                }
                                virtual={undefined}
                            >
                                Share it
                            </Styled(Text)>
                        </Styled(Left)>
                    </Styled(Button)>
                </Styled(Body)>
            </Styled(CardItem)>
        </Styled(Card)>
    </Styled(Content)>
</Styled(Container)>,
  "nodes": Array [
    <Styled(Container)
      virtual={undefined}
>
      <Styled(Content)
            virtual={undefined}
      >
            <Styled(Card)
                  style={
                        Object {
                              "flex": 0,
                            }
                  }
                  virtual={undefined}
            >
                  <Styled(CardItem)
                        virtual={undefined}
                  >
                        <Styled(Left)
                              virtual={undefined}
                        >
                              <Styled(Thumbnail)
                                    source={
                                          Object {
                                                "uri": "https://media.glassdoor.com/sql/515676/zomato-squarelogo-1466750863527.png",
                                              }
                                    }
                                    style={
                                          Object {
                                                "height": 50,
                                                "width": 50,
                                              }
                                    }
                                    virtual={undefined}
                              />
                              <Styled(Body)
                                    virtual={undefined}
                              >
                                    <Styled(Text)
                                          virtual={undefined}
                                    />
                                    <Styled(Text)
                                          note={true}
                                          virtual={undefined}
                                    >
                                          04 May 2017
                                    </Styled(Text)>
                              </Styled(Body)>
                        </Styled(Left)>
                  </Styled(CardItem)>
                  <Styled(CardItem)
                        virtual={undefined}
                  >
                        <Styled(Body)
                              virtual={undefined}
                        >
                              <Image
                                    source={
                                          Object {
                                                "uri": undefined,
                                              }
                                    }
                                    style={
                                          Object {
                                                "height": 200,
                                                "width": 340,
                                              }
                                    }
                              />
                              <Styled(Text)
                                    virtual={undefined}
                              />
                              <Styled(Button)
                                    textStyle={
                                          Object {
                                                "color": "#87838B",
                                              }
                                    }
                                    transparent={true}
                                    virtual={undefined}
                              >
                                    <Styled(Icon)
                                          name="logo-facebook"
                                          virtual={undefined}
                                    />
                                    <Styled(Left)
                                          virtual={undefined}
                                    >
                                          <Styled(Text)
                                                style={
                                                      Object {
                                                            "fontSize": 12,
                                                            "marginLeft": 2,
                                                          }
                                                }
                                                virtual={undefined}
                                          >
                                                Share it
                                          </Styled(Text)>
                                    </Styled(Left)>
                              </Styled(Button)>
                        </Styled(Body)>
                  </Styled(CardItem)>
            </Styled(Card)>
      </Styled(Content)>
</Styled(Container)>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <DetailCollection
        navigation={
                Object {
                        "state": Object {
                          "params": Object {
                            "collection": Object {},
                          },
                        },
                      }
        }
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": DetailCollection {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "navigation": Object {
            "state": Object {
              "params": Object {
                "collection": Object {},
              },
            },
          },
        },
        "refs": Object {},
        "state": Object {},
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Styled(Container)
          virtual={undefined}
>
          <Styled(Content)
                    virtual={undefined}
          >
                    <Styled(Card)
                              style={
                                        Object {
                                                  "flex": 0,
                                                }
                              }
                              virtual={undefined}
                    >
                              <Styled(CardItem)
                                        virtual={undefined}
                              >
                                        <Styled(Left)
                                                  virtual={undefined}
                                        >
                                                  <Styled(Thumbnail)
                                                            source={
                                                                      Object {
                                                                                "uri": "https://media.glassdoor.com/sql/515676/zomato-squarelogo-1466750863527.png",
                                                                              }
                                                            }
                                                            style={
                                                                      Object {
                                                                                "height": 50,
                                                                                "width": 50,
                                                                              }
                                                            }
                                                            virtual={undefined}
                                                  />
                                                  <Styled(Body)
                                                            virtual={undefined}
                                                  >
                                                            <Styled(Text)
                                                                      virtual={undefined}
                                                            />
                                                            <Styled(Text)
                                                                      note={true}
                                                                      virtual={undefined}
                                                            >
                                                                      04 May 2017
                                                            </Styled(Text)>
                                                  </Styled(Body)>
                                        </Styled(Left)>
                              </Styled(CardItem)>
                              <Styled(CardItem)
                                        virtual={undefined}
                              >
                                        <Styled(Body)
                                                  virtual={undefined}
                                        >
                                                  <Image
                                                            source={
                                                                      Object {
                                                                                "uri": undefined,
                                                                              }
                                                            }
                                                            style={
                                                                      Object {
                                                                                "height": 200,
                                                                                "width": 340,
                                                                              }
                                                            }
                                                  />
                                                  <Styled(Text)
                                                            virtual={undefined}
                                                  />
                                                  <Styled(Button)
                                                            textStyle={
                                                                      Object {
                                                                                "color": "#87838B",
                                                                              }
                                                            }
                                                            transparent={true}
                                                            virtual={undefined}
                                                  >
                                                            <Styled(Icon)
                                                                      name="logo-facebook"
                                                                      virtual={undefined}
                                                            />
                                                            <Styled(Left)
                                                                      virtual={undefined}
                                                            >
                                                                      <Styled(Text)
                                                                                style={
                                                                                          Object {
                                                                                                    "fontSize": 12,
                                                                                                    "marginLeft": 2,
                                                                                                  }
                                                                                }
                                                                                virtual={undefined}
                                                                      >
                                                                                Share it
                                                                      </Styled(Text)>
                                                            </Styled(Left)>
                                                  </Styled(Button)>
                                        </Styled(Body)>
                              </Styled(CardItem)>
                    </Styled(Card)>
          </Styled(Content)>
</Styled(Container)>,
        "_debugID": 2,
        "_renderedOutput": <Styled(Container)
          virtual={undefined}
>
          <Styled(Content)
                    virtual={undefined}
          >
                    <Styled(Card)
                              style={
                                        Object {
                                                  "flex": 0,
                                                }
                              }
                              virtual={undefined}
                    >
                              <Styled(CardItem)
                                        virtual={undefined}
                              >
                                        <Styled(Left)
                                                  virtual={undefined}
                                        >
                                                  <Styled(Thumbnail)
                                                            source={
                                                                      Object {
                                                                                "uri": "https://media.glassdoor.com/sql/515676/zomato-squarelogo-1466750863527.png",
                                                                              }
                                                            }
                                                            style={
                                                                      Object {
                                                                                "height": 50,
                                                                                "width": 50,
                                                                              }
                                                            }
                                                            virtual={undefined}
                                                  />
                                                  <Styled(Body)
                                                            virtual={undefined}
                                                  >
                                                            <Styled(Text)
                                                                      virtual={undefined}
                                                            />
                                                            <Styled(Text)
                                                                      note={true}
                                                                      virtual={undefined}
                                                            >
                                                                      04 May 2017
                                                            </Styled(Text)>
                                                  </Styled(Body)>
                                        </Styled(Left)>
                              </Styled(CardItem)>
                              <Styled(CardItem)
                                        virtual={undefined}
                              >
                                        <Styled(Body)
                                                  virtual={undefined}
                                        >
                                                  <Image
                                                            source={
                                                                      Object {
                                                                                "uri": undefined,
                                                                              }
                                                            }
                                                            style={
                                                                      Object {
                                                                                "height": 200,
                                                                                "width": 340,
                                                                              }
                                                            }
                                                  />
                                                  <Styled(Text)
                                                            virtual={undefined}
                                                  />
                                                  <Styled(Button)
                                                            textStyle={
                                                                      Object {
                                                                                "color": "#87838B",
                                                                              }
                                                            }
                                                            transparent={true}
                                                            virtual={undefined}
                                                  >
                                                            <Styled(Icon)
                                                                      name="logo-facebook"
                                                                      virtual={undefined}
                                                            />
                                                            <Styled(Left)
                                                                      virtual={undefined}
                                                            >
                                                                      <Styled(Text)
                                                                                style={
                                                                                          Object {
                                                                                                    "fontSize": 12,
                                                                                                    "marginLeft": 2,
                                                                                                  }
                                                                                }
                                                                                virtual={undefined}
                                                                      >
                                                                                Share it
                                                                      </Styled(Text)>
                                                            </Styled(Left)>
                                                  </Styled(Button)>
                                        </Styled(Body)>
                              </Styled(CardItem)>
                    </Styled(Card)>
          </Styled(Content)>
</Styled(Container)>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <DetailCollection
    navigation={
        Object {
            "state": Object {
              "params": Object {
                "collection": Object {},
              },
            },
          }
    }
/>,
}
`;
